front/lib/api/files/upload.ts

```diff
@@ -387,7 +388,10 @@ const maybeApplyProcessing: ProcessingFunction = async (
 
 export async function processAndStoreFile(
   auth: Authenticator,
-  { file, req }: { file: FileResource; req: IncomingMessage }
+  {
+    file,
+    reqOrString,
```

Comment:
Can we use a discriminant union here? We are lucky enough to have TS if we can leverage it all the way 🙏

Reply:
More tokens but I don't quite see the added value here? We'd write a type guard instead of `typeof reqOrString === "string"` based on the internal type of request but all in all that would not provide any additional safety would it?

Reply:
Either a typeguard or two distinct argument types. My point is that typeof === "string" is so JavaScript 🙃 for this use case.